internal/test/try/try_test.go

```diff
@@ -132,3 +134,85 @@ func TestEventually(t *testing.T) {
 		})
 	}
 }
+
+// TODO(wallrj): Add tests for failure cases.
+// For this I searched for a way to modify / wrap `testing.T` so that it fails
+// if the there are no assertion errors, and succeeds if there are errors, with
+// some way to check the assertion message.
```

Comment:
consider changing the signature of `CheckStructFields` to be
```
CheckStructFields(expectations map[string]interface{}, actual interface{}) error
```
and then make assertions on the error